File: data/src/main/java/com/ykb/app/cryptotrader/data/model/Strategy.java
package com.ykb.app.cryptotrader.data.model;

import com.ykb.app.cryptotrader.utils.enums.StrategyNames;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "STRATEGY")
@Data
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Strategy extends BaseIdEntity {

    @Column(name = "NAME", columnDefinition = "VARCHAR(256)", nullable = false)
    @Enumerated(EnumType.STRING)
    private StrategyNames name;

    @ManyToOne
    @JoinColumn(name = "SETTINGS_ID", nullable = false)
    private StrategySettings settings;

}

File: library/data/data-domain/src/main/kotlin/com/ykb/app/cryptotrader/data/domain/model/Strategy.kt
package com.ykb.app.cryptotrader.data.domain.model

import com.ykb.app.cryptotrader.data.base.entity.BaseIdEntity
import com.ykb.app.cryptotrader.domain.enums.StrategyNames
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "STRATEGY")
class Strategy(
    @Column("NAME", columnDefinition = "VARCHAR(256)")
    @Enumerated(EnumType.STRING)
    val name: StrategyNames,

    @ManyToOne
    @JoinColumn(name = "SETTINGS_ID", nullable = false)
    private val settings: StrategySettings
) : BaseIdEntity()